File: e2e/cypress-javascript/providers/StepProvider.js
const { AccessibilityStep } = require('@/interactions/steps/AccessibilityStep');
const { CreateTaskStep } = require('@/interactions/steps/CreateTaskStep');
const { EditTaskStep } = require('@/interactions/steps/EditTaskStep');
const { LanguageSwitcherStep } = require('@/interactions/steps/LanguageSwitcherStep');
const { NavigationStep } = require('@/interactions/steps/NavigationStep');
const { TaskTableStep } = require('@/interactions/steps/TaskTableStep');

class StepProvider {
  constructor() {
    this.navigation = new NavigationStep();
    this.tasks = new TaskTableStep();
    this.createTask = new CreateTaskStep();
    this.editTask = new EditTaskStep();
    this.language = new LanguageSwitcherStep();
    this.accessibility = new AccessibilityStep();
  }
}

module.exports = { StepProvider };

File: e2e/capybara-ruby/providers/step_provider.rb
# frozen_string_literal: true

require_relative '../interactions/steps/accessibility_step'
require_relative '../interactions/steps/create_task_step'
require_relative '../interactions/steps/edit_task_step'
require_relative '../interactions/steps/language_switcher_step'
require_relative '../interactions/steps/navigation_step'
require_relative '../interactions/steps/task_table_step'

class StepProvider
  attr_reader :navigation, :tasks, :create_task, :edit_task, :language, :accessibility

  def initialize
    @navigation = NavigationStep.new
    @tasks = TaskTableStep.new
    @create_task = CreateTaskStep.new
    @edit_task = EditTaskStep.new
    @language = LanguageSwitcherStep.new
    @accessibility = AccessibilityStep.new
  end
end
